packages/@aws-cdk/cloudformation-include/test/invalid-templates.test.ts

```diff
@@ -100,6 +100,30 @@ describe('CDK Include', () => {
       includeTestTemplate(stack, 'output-referencing-nonexistant-condition.json');
     }).toThrow(/Output with name 'SomeOutput' refers to a Condition with name 'NonexistantCondition' which was not found in this template/);
   });
+
+  test("throws a validation exception when Fn::Sub in string form uses a key that isn't in the template", () => {
+    expect(() => {
+      includeTestTemplate(stack, 'fn-sub-key-not-in-template-string.json');
+    }).toThrow(/Element used in Ref expression with logical ID: 'AFakeResource' in Fn::Sub not found/);
+  });
+
+  test("throws a validation exception when Fn::Sub in map form uses a key that isn't in the template", () => {
+    expect(() => {
+      includeTestTemplate(stack, 'fn-sub-key-not-in-template.json');
+    }).toThrow(/Element used in Ref expression with logical ID: 'AFakeResource' not found/);
+  });
+
+  test("throws a validation exception when Fn::Sub in map form uses a key that isn't in the template and isn't in the map", () => {
+    expect(() => {
+      includeTestTemplate(stack, 'fn-sub-key-not-in-map.json');
+    }).toThrow(/Element used in Ref expression with logical ID: 'AFakeResource' in Fn::Sub not found/);
+  });
+
+  test("throws a validation exception when Fn::Sub uses a resource attribute that doesn't exist", () => {
+    expect(() => {
+      includeTestTemplate(stack, 'fn-sub-bad-get-att.json');
+    }).toThrow(/Resource referenced in Fn::Sub expression with logical ID: 'FakeBucket' not found/);
+  });
```

Comment:
More tests are needed here I think:

1. `Fn::Sub` with "$".
2. `Fn::Sub` with "${".
3. `Fn::Sub` with "${}".
4. `Fn::Sub` with "$}".
5. `Fn::Sub` with "}".

I don't know what the CloudFormation behavior is for a lot of these, so that should be verified as well.

Reply:
Do you mean invocations like

1.  "Fn::Sub": "$" 
2. "Fn::Sub": "${"

or do you mean

1. "Fn::Sub": "$SomeResource" 
2. "Fn::Sub": "${SomeResource"

Reply:
Both :)

Reply:
turns out all of those except #3 are valid, and none of them (expect 3) are interpreted as references.